### PR TITLE
Refactor: general tweaks to adjust the core to the recent changes implemented in the Authorize.Net gateway addon

### PR DESCRIPTION
**Pull request by glaubersilva**
## Description

**This PR implements the following:**

- A new React component to be used on the Donor Dashboard when the user updates the payment method (credit card) for Authorize subscriptions. Now instead of sending the raw Credit Card data to the backend, we send the "data descriptor" and "data value" params generated by the [Accept.Js](https://developer.authorize.net/api/reference/features/acceptjs.html) library.

**Beyond the Authorize.Net gateway, the items below will benefit other gateways as well:**

- A way to handle update errors, on the donor dashboard, thrown by the backend during the update process.

- Changes the way that the form classic template adds the gateway field in the DOM, now it uses jQuery, and this way if any scripts are present with the fields, they also will be loaded.

- Adds a new donor email notification for donations with the "processing" status + notice on the "thank you" page informing the user that a final receipt will be sent to the email when the donation is completed.

## Affects

- Subscriptions created with Authorize.Net using the new gateway API and the Accept.js library
- Email Notifications
- Donor Dashboard
- Donation Receipt

## Visuals

image

image

image

image

image

## Testing Instructions

First, pull the following PRs and setup the Authorize gateway:

https://github.com/impress-org/give-authorize-gateway/pull/111

https://github.com/impress-org/give-recurring/pull/1156

Second, create Authorize subscription using [a test credit card](https://developer.authorize.net/hello_world/testing_guide.html) and check if the thank you page has a notice just below the donation status and also if the "Donation Processing Receipt" notification was sent to the email. After that go to the donor dashboard and edit the credit card data to another one.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [x] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203580457557897

### CODE REVIEW

**Review by jonwaldstein — APPROVED** [review]
@glaubersilva I was not able to test this locally but the logic looks correct.  Please send this off to QA to confirm functionality.

**Review by Benunc — APPROVED** [review]
This works flawlessly. I tested with a subscription previously made, and it changed the payment method with no issue.

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/Email/Notifications/DonationProcessingReceipt.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                return;
+            }
+
+            $this->recipient_email = $this->payment->email;
+
+            $this->send_email_notification(
+                [
+                    'payment_id' => $this->payment->ID,
+                ]
+            );
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/Email/Notifications/DonationProcessingReceipt.php]: Resolved in bed80a979e43401fa30ef063e8c3e0872eed94b1

File: src/Email/Notifications/DonationProcessingReceipt.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    /* @var Give_Payment $payment */
+    public $payment;
+
+    /**
+     * @unreleased
+     */
+    public function init()
+    {
+        // Initialize empty payment.
+        $this->payment = new Give_Payment(0);
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/Email/Notifications/DonationProcessingReceipt.php]: Resolved in bed80a979e43401fa30ef063e8c3e0872eed94b1

**Comment by glaubersilva** [comment]
@jonwaldstein Thanks for the improvement suggestion, already implemented it! Let me know if there is anything more. ;)

File: src/Email/Notifications/DonationProcessingReceipt.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+use Give_Email_Notification;
+use Give_Payment;
+
+/**
+ * @unreleased
+ */
+class DonationProcessingReceipt extends Give_Email_Notification
+{
+    /* @var Give_Payment $payment */
+    public $payment;
```
Comment (jonwaldstein) [inline src/Email/Notifications/DonationProcessingReceipt.php:16]: @glaubersilva you can remove this property too since its not used

Comment (glaubersilva) [reply src/Email/Notifications/DonationProcessingReceipt.php]: Resolved in 8cab5f4989eecfd72b96e15e1433cee7c11abef7

**Comment by glaubersilva** [comment]
@jonwaldstein Thanks! I missed it before but is already been removed.

**Review by jonwaldstein — APPROVED** [review]
@glaubersilva great job!

**Review by Benunc — APPROVED** [review]
The new email has been tested, and it works. Everything looks to be all set on this. Passes all automated tests as well.

**Pull request merged**